src/actors/upgrade/network_status/actors/netifacescanner/actor.py

```diff
@@ -0,0 +1,125 @@
+        iface_stats = subprocess.Popen(('ethtool', '-i', iface),
+                                       stdout=subprocess.PIPE, stderr=devnull)
+        iface_stats, err = iface_stats.communicate()
+        driver = re.search(r"(driver):\s+(.*)$", iface_stats,
+                           flags=re.MULTILINE)
+        if driver is not None:
+            driver = driver.group(2)
+        else:
+            driver = "None"
+        return driver
+
+    def return_ifs_info(self):
+        sys_command = ['find', '/sys/devices/', '-name', 'net', '-type', 'd']
+        if_names = self.get_ifaces_names(sys_command)
+        journal = subprocess.Popen(('journalctl', '-xe', '--no-tail'), stdout=subprocess.PIPE)
+        journal, err = journal.communicate()
+        ifs_info = []
+        result = IfaceResult()
+        for if_name in if_names:
+            with open(os.devnull, 'w') as devnull:
+                driver = self.get_net_driver(if_name, devnull)
+                ethinf = subprocess.Popen(('ethtool', '-P', if_name),
+                                          stdout=subprocess.PIPE, stderr=devnull)
+            ethinf, err = ethinf.communicate()
+            hwaddr = ethinf.split()[2]
+            network_script = self.get_conf_file(if_name)
+            search_string = r".*\(" + if_name + r"\):\s+address\s+(.*)"
+            ipv4_string = re.compile(r"^(IPADDR)\s*=\s*\"?(.*)\"?$")
+            boot_string = re.compile(r"^(BOOTPROTO)\s*=\s*\"?(.*)\"?$")
+            boot_proto = self.grep_last_occurence(boot_string, network_script)
+            if isinstance(boot_proto, list):
+                boot_proto = boot_proto[-1][1]
+            if boot_proto.lower() == "dhcp":
+                ip = self.grep_ips(search_string, journal)
+                net_if_info = [hwaddr, driver, ip]
+                ifs_info.append(net_if_info)
+            else:
+                ip = self.grep_last_occurence(ipv4_string, network_script)
+                if isinstance(ip, list):
+                    ip = ip[-1][1]
+                else:
+                    ip = "None"
+                net_if_info = [hwaddr, driver, ip]
+                result.items.append(IfacesInfo(
+                    if_name=if_name,
+                    hwaddr=hwaddr,
+                    driver=driver,
+                    ipv4addr=ip))
+                ifs_info.append(net_if_info)
+            runtime_hw = self.grep_runtime_hws(if_name)
+            if runtime_hw:
+                if runtime_hw != hwaddr:
+                    self.log.info("The persistent hwaddr of interface %s is %s, but runtime hwaddr is %s"
+                                  % (if_name, hwaddr, runtime_hw))
+        return result
+
+    def get_conf_file(self, file_name):
+        script_dir = "/etc/sysconfig/network-scripts"
+        confile = "ifcfg-" + file_name
+        if (confile) in os.listdir(script_dir):
+            return os.path.join(script_dir, confile)
+
+    def grep_ips(self, pattern, data):
```

Comment:
Regarding to the use, would be much better to require the `if_name` as the parameter instead of the `pattern` which you can compose inside. Rename function to something like `get_ifname_ip`. I guess you return always only one IP right? In case it would containt more IPs, `get_ifname_ips`. (Or whatever name you want to use, but `get_` is better then grep.